File: parkway_walk.py
from sys import stdin, stdout

t = int(stdin.readline())

for _ in range(t):
  n, m = map(int, stdin.readline().strip().split())
  arr = list(map(int, stdin.readline().strip().split()))

  ans = 0
  if n > 1:
    for i in range(n-1):
      m -= arr[i]
      if arr[i + 1] > m:
        r = arr[i + 1] - m
        m += r
        ans += r
  else:
    ans = abs(m - arr[0])

  stdout.write(str(ans))
  stdout.write('\n')

File: parkway_walk.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);

  int t;
  cin >> t;

  while (t--) {
    int n, m;
    cin >> n >> m;

    int arr[n];
    for (int i = 0; i < n; i++) {
      cin >> arr[i];
    }

    int ans = 0;
    if (n > 1) {
      for (int i = 0; i < n - 1; i++) {
        m -= arr[i];
        if (arr[i + 1] > m) {
          int r = arr[i + 1] - m;
          m += r;
          ans += r;
        }
      }
    } else
      ans = abs(m - arr[0]);

    cout << ans << '\n';
  }
}